svc-mesh/service-mesh.md

```diff
@@ -0,0 +1,251 @@
+the service. That's how the interaction between the control plane and the
+proxy allows the user to apply load balancing and authentication rules to
+the incoming and outgoing traffic, inside the cluster and between multiple
+micro-services.
+
+This could not happen without a good amount of `iptables` rules ensuring
+the packets reach the proxy instead of the expected service. Rules are
+setup through an __init__ container because they have to be there as soon
+as the proxy starts.
+
+## Prerequisites
+
+### Kata and Kubernetes
+
+Follow the [instructions](https://github.com/kata-containers/documentation/blob/master/install/README.md)
+to get Kata Containers properly installed and configured with Kubernetes.
+You can choose between CRI-O and CRI-containerd, both will be supported
+through this document.
+
+For both cases, select the workloads as _trusted_ by default. This way,
+your cluster and your service mesh will run with `runc`, and only the
+containers you choose to annotate will run with Kata Containers.
+
+### Restrictions
+
+As documented [here](https://github.com/runconduit/conduit/issues/982),
+a kernel version between 4.14.22 and 4.14.40 will cause a deadlock when
+`getsockopt()` get called with `SO_ORIGINAL_DST` option. Unfortunately,
+both service meshes use this system call with this same option from the
+proxy container running inside the VM. This means that you cannot run
+this kernel version range as the guest kernel for Kata if you want your
+service mesh to work.
+
+As mentioned when explaining the basic functioning of those service meshes,
+`iptables` are heavily used, and they need to be properly enabled through
+the guest kernel config, otherwise the __init__ container will not be able
+to perform a proper setup of the rules.
+
+## Install and deploy your service mesh
+
+### Istio
+
+As a reference, you can follow Istio [instructions](https://istio.io/docs/setup/kubernetes/quick-start/#download-and-prepare-for-the-installation).
+
+Here is a summary of what you need to install Istio on your system
+```
+$ curl -L https://git.io/getLatestIstio | sh -
+$ cd istio-*
+$ export PATH=$PWD/bin:$PATH
+```
+
+Let's now deploy Istio in the control plane of your cluster
+```
+$ kubectl apply -f install/kubernetes/istio-demo.yaml
+```
+
+In order to verify the control plane has been properly deployed, you
+can use both of those following commands
+```
+$ kubectl get svc -n istio-system
+$ kubectl get pods -n istio-system -o wide
+```
+
+### Conduit
+
+As a reference, you can follow Conduit [instructions](https://conduit.io/getting-started/index.html).
+
+Here is a summary of what you need to install Conduit on your system
+```
+$ curl https://run.conduit.io/install | sh
+$ export PATH=$PATH:$HOME/.conduit/bin
+```
+
+Let's now deploy Conduit in the control plane of your cluster
+```
+$ conduit install | kubectl apply -f -
+```
+
+In order to verify the control plane has been properly deployed, you
```

Comment:
To verify that the control plane is properly deployed, you can use both of the following commands:

Reply:
done